bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/dto/ThingMessageDTO.java

```diff
@@ -0,0 +1,230 @@
+
+/**
+ * Superclass for all Thing message types.
+ * 
+ * @author Ronny Grun - Initial contribution
+ */
+public class ThingMessageDTO {
+
+    @SerializedName("type")
+    @Expose
+    private String type;
+    @SerializedName("channelType")
+    @Expose
+    private String channelType;
+    @SerializedName("uom")
+    @Expose
+    private String uom;
+    @SerializedName("value")
+    @Expose
+    private String value;
+    @SerializedName("feature")
+    @Expose
+    private String feature;
+    @SerializedName("featureClear")
+    @Expose
+    private String featureClear;
+    @SerializedName("featureName")
+    @Expose
+    private String featureName;
+    @SerializedName("featureDescription")
+    @Expose
+    private String featureDescription;
+    @SerializedName("deviceId")
+    @Expose
+    private String deviceId;
+    @SerializedName("commands")
+    @Expose
+    private FeatureCommands commands;
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public void setChannelType(String channelType) {
+        this.channelType = channelType;
+    }
+
+    public String getChannelType() {
+        return channelType;
+    }
+
+    public String getUom() {
+        return uom;
+    }
+
+    public void setUom(String uom) {
+        this.uom = uom;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public void setValue(String value) {
+        this.value = value;
+    }
+
+    public String getFeature() {
+        return feature;
+    }
+
+    public void setFeature(String feature) {
+        this.feature = feature;
+    }
+
+    public String getFeatureClear() {
+        return featureClear;
+    }
+
+    public void setFeatureClear(String featureClear) {
+        this.featureClear = featureClear;
+    }
+
+    public String getFeatureName() {
+        return featureName;
+    }
+
+    public void setFeatureName(String featureName) {
+        this.featureName = featureName;
+    }
+
+    public String getFeatureDescription() {
+        return featureDescription;
+    }
+
+    public void setFeatureDescription(String featureDescription) {
+        this.featureDescription = featureDescription;
+    }
+
+    public String getDeviceId() {
+        return deviceId;
+    }
+
+    public void setDeviceId(String deviceId) {
+        this.deviceId = deviceId;
+    }
+
+    public FeatureCommands getCommands() {
+        return commands;
+    }
+
+    public void setCommands(FeatureCommands commands) {
+        this.commands = commands;
+    }
+
+    public String getChannelId() {
+        StringBuilder sb = new StringBuilder();
+        String f = feature.replace(".", ";");
+        String parts[] = f.split(";");
+        int count = 0;
+        for (String str : parts) {
+            if (count != 0) {
+                sb.append(str.substring(0, 1).toUpperCase() + str.substring(1));
```

Comment:
```suggestion
                sb.append(str.substring(0, 1).toUpperCase()).append(str.substring(1)));
```

avoid string concatenation if you have the builder